### ISSUE TITLE
[hooks] Support environment variables for command hooks

### ISSUE THREAD

**Issue by crschnick** [jreleaser/jreleaser#1351]
Would it be possible to pass all relevant information with https://jreleaser.org/guide/latest/reference/hooks/command.html through environment variables? Passing everything as arguments is quite tedious as that would result in like ten arguments in my case as I pass all the project information.

**Comment by aalmiray** [comment]
Execution of external commands should inherit the current environment used by the JReleaser process.
Yet, any model values should be defined using name templates such as `{{projectVersion}}. Unfortunately for command hooks, name template substitution only applies to the command line because the invoked command may be a plain binary and not a script.

For this reason JReleaser will support script hooks in the next release https://jreleaser.org/guide/early-access/reference/hooks/script.html

There is however, the option to expose specific model properties as environment variables before running a command. Mapping the whole model would be troublesome, thus marking which values you're interested in mapping would be better. For example:

```yml
hooks:
  command:
    before:
      - cmd: 'doThething'
        environment:
          PROJECT_VERSION: '{{projectVersion}}'
          BREW_ACTIVE: '{{ Model.distributions.app.brew.active }}'
```

The `Model` context does not yet exist. I wonder how easily it could be done given that it pretty much requires performing an object graph search.

**Comment by crschnick** [comment]
> 
> ```yaml
> hooks:
>   command:
>     before:
>       - cmd: 'doThething'
>         environment:
>           PROJECT_VERSION: '{{projectVersion}}'
>           BREW_ACTIVE: '{{ Model.distributions.app.brew.active }}'
> ```
> 

Yes, I had something like this in mind. The model context would be very useful in some cases.

**Comment by aalmiray** [comment]
FWIW I think Mustache can traverse the context OOTB. Will run some tests.

Thus RFE proposes the following changes:

- a `Model` context for evaluating model properties in [Mustache] name templates.
- an `environment` property added to `CommandHook` and `ScriptHook.`

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.6.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.6.0))
